libs/parallelism/executors/include/hpx/executors/execution_policy_annotation.hpp

```diff
@@ -0,0 +1,49 @@
+//  Copyright (c) 2021 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+/// \file hpx/execution/execution_policy_annotation.hpp
+
+#pragma once
+
+#include <hpx/config.hpp>
+#include <hpx/execution/executors/execution_parameters.hpp>
+#include <hpx/execution/executors/rebind_executor.hpp>
+#include <hpx/execution/traits/is_execution_policy.hpp>
+#include <hpx/properties/property.hpp>
+
+#include <type_traits>
+#include <utility>
+
+namespace hpx { namespace execution { namespace experimental {
+
+    // make_with_annotation property implementation for execution policies
+    // that simply forwards to the embedded executor
+    template <typename ExPolicy,
+        typename Enable =
+            std::enable_if_t<hpx::is_execution_policy<ExPolicy>::value>>
+    constexpr decltype(auto) tag_invoke(
+        hpx::execution::experimental::make_with_annotation_t, ExPolicy&& policy,
+        char const* annotation)
+    {
+        auto exec = hpx::experimental::prefer(
+            hpx::execution::experimental::make_with_annotation,
+            policy.executor(), annotation);
```

Comment:
What I had in mind was something like this:

```suggestion
    template <typename ExPolicy,
        typename Enable =
            std::enable_if_t<hpx::is_execution_policy<ExPolicy>::value &&
            hpx::is_invocable<hpx::execution::experimental::make_with_annotation_t,
                typename ExPolicy::executor_type, char const*>::value>>
    constexpr decltype(auto) tag_invoke(
        hpx::execution::experimental::make_with_annotation_t, ExPolicy&& policy,
        char const* annotation)
    {
        auto exec = hpx::execution::experimental::make_with_annotation(
            policy.executor(), annotation);
```

`can_prefer` could probably be useful in some contexts (we don't have it), but would almost always be true (I don't know yet which properties would be useful to never ignore if they can't be applied). I also don't think `can_prefer` would be right in this context.

`can_require` is doable, but kind of weird, because it would essentially be `template <typename Tag, typename... Args> can_require = hpx::is_invocable<Tag, Args...>::value`. I'm not sure if there's any sensible way to constrain it, but if not, I don't see a point in that kind of an alias.

I might be overseeing something though...

Reply:
Makes sense. I'll go with `is_tag_invocable` instead of the plain `is_invocable`, though

Reply:
Makes sense! Forgot about that one.